### PR TITLE
[8.15] Fix DLS using runtime fields and synthetic source (#112341)

### PR DESCRIPTION
**Pull request by original-brownbear**
Backports the following commits to 8.15:
 - Fix DLS using runtime fields and synthetic source (#112341)

### CODE REVIEW

**Comment by javanna** [comment]
Have we decided not to get this fix in?

**Comment by original-brownbear** [comment]
@elasticsearchmachine update branch

**Comment by original-brownbear** [comment]
Urgh sorry @javanna this failed to auto-merge for some reason and I missed it, let me try again.

**Comment by original-brownbear** [comment]
@elasticmachine update branch

**Comment by drempapis** [comment]
@elasticmachine test this please

**Comment by drempapis** [comment]
@elasticmachine update branch

**Comment by drempapis** [comment]
@elasticmachine update branch

**Comment by drempapis** [comment]
@elasticmachine update branch

**Comment by drempapis** [comment]
@elasticmachine test this please

**Comment by drempapis** [comment]
@elasticmachine test this please

**Comment by drempapis** [comment]
Version 8.15 is no longer maintained; therefore, this issue is now closed.
